src/main/java/com/mineinabyss/mobzy/mobs/types/ProjectileEntity.kt

```diff
@@ -0,0 +1,36 @@
+package com.mineinabyss.mobzy.mobs.types
+
+import com.mineinabyss.geary.ecs.components.get
+import com.mineinabyss.geary.ecs.engine.Engine
+import com.mineinabyss.mobzy.api.nms.aliases.NMSEntityType
+import com.mineinabyss.mobzy.api.nms.aliases.NMSSnowball
+import com.mineinabyss.mobzy.api.nms.aliases.NMSWorld
+import com.mineinabyss.mobzy.ecs.components.initialization.ItemModel
+import com.mineinabyss.mobzy.mobs.CustomEntity
+import net.minecraft.server.v1_16_R2.EntitySnowball
+import net.minecraft.server.v1_16_R2.EntityTypes
+import net.minecraft.server.v1_16_R2.MovingObjectPosition
+import net.minecraft.server.v1_16_R2.MovingObjectPositionEntity
+import org.bukkit.craftbukkit.v1_16_R2.inventory.CraftItemStack
+
+class ProjectileEntity(
+    type: NMSEntityType<*>,
+    world: NMSWorld
+) : EntitySnowball(type as EntityTypes<EntitySnowball>, world), CustomEntity {
+    override val gearyId: Int = Engine.getNextId()
+
+    override val nmsEntity: NMSSnowball get() = this
+
+    init {
+        initEntity()
+        item = CraftItemStack.asNMSCopy(get<ItemModel>()?.item?.toItemStack())
+    }
+
+    override fun a(var0: MovingObjectPositionEntity) {
+        return
+    }
+
+    override fun a(var0: MovingObjectPosition) {
+        return
```

Comment:
How come we're stopping these from running? It seems to just broadcast its death? Feels like stopping this might make the entity not get removed on collision or maybe some weird stuff on client.

Also I think I prefer `fun whatever() = Unit` when you explicitly want a function to do nothing